src/program.cpp

```diff
@@ -221,15 +233,42 @@ std::vector<argument> generic_eval(const module& p,
                     assert(results.find(i) != results.end());
                     return results[i];
                 });
-            results.emplace(ins, trace(ins, [&] {
-                                return ins->normalized_operator().compute(
-                                    ctx, ins->get_shape(), values);
-                            }));
+
+            const auto& mod_args = ins->module_inputs();
+            if(not mod_args.empty())
+                results.emplace(
+                    ins, trace(ins, mod, [&] {
+                        return ins->get_operator().compute(
+                            values,
+                            mod_args,
+                            [&](module_ref smod,
```

Comment:
This lambda can be assigned to a variable and then passed in. I think it will be cleaner.

Reply:
fixed accordingly